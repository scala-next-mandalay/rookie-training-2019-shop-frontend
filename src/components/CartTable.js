import React from 'react';
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import CardMedia from '@material-ui/core/CardMedia';
import { BASEURL_ITEM_IMAGES } from '../constants'
import Box from '@material-ui/core/Box'
import Button from '@material-ui/core/Button';
import DeleteIcon from '@material-ui/icons/Delete';
import QuantitySelect from '../containers/QuantitySelect'

const useStyles = makeStyles(theme => ({
  container: {
    overflowX: 'auto',
    marginRight: 'auto',
    marginLeft: 'auto',
    marginTop: theme.spacing(2),
    padding: theme.spacing(1),
    margin: theme.spacing(1),
  },
  tableCellData: {
    textAlign: 'center',
  },
  tableCellHeader: {
    textAlign: 'left',
  },
  thumbnail: {
    width: 75,
    height: 96
  },
  card: {
    display: 'flex',
  },
  cardTitle: {
    display: 'flex',
    flexDirection: 'column',
    margin: 'auto',
    marginLeft: theme.spacing(2),
  },
  rightIcon: {
    marginLeft: theme.spacing(1),
  },
}));

const CartTable = ({ cart, deleteCartItem }) => {
  const classes = useStyles()

  const tableRows = []
  for (const row of cart) {
    tableRows.push(
      <TableRow key={row.id}>
        <TableCell>
          <div className={classes.card}>
            <CardMedia
              className={classes.thumbnail}
              image={BASEURL_ITEM_IMAGES+row.image}
              //image={"http://127.0.0.1/dummyImage.jpg"}
              title={row.name}
            />
            <Box className={classes.cardTitle}>
              {row.name}
            </Box>
          </div>
        </TableCell>
        <TableCell>{row.price}</TableCell>
        <TableCell>
          <QuantitySelect id={row.id} quantity={row.quantity} />          
        </TableCell>
        <TableCell>{row.price * row.quantity}</TableCell>     
        <TableCell>
          <Button 
            variant="contained" 
            size="small" 
            className={classes.button}
            onClick={()=> deleteCartItem(row.id)}
          >
            Delete
            <DeleteIcon className={classes.rightIcon} />
          </Button>         
        </TableCell>     
      </TableRow>
    )
  }

  return (
    <Paper className="container">
      <Table className={classes.table}>
        <TableHead>
          <TableRow>
            <TableCell>Product</TableCell>
            <TableCell>Price (Ks)</TableCell>
            <TableCell>Qty</TableCell>
            <TableCell>Total (Ks)</TableCell>
            <TableCell>Action</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>{tableRows}</TableBody>
      </Table>
    </Paper>
  )
}

CartTable.propTypes = {
  cart: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    image: PropTypes.string,
    quantity: PropTypes.number.isRequired,
  }).isRequired).isRequired,
  deleteCartItem: PropTypes.func,
}

export default CartTable;
