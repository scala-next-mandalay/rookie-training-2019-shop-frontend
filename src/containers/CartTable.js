import { connect } from 'react-redux';
import CartTable from '../components/CartTable';

export default connect(
  (state) => ({
    cart: state.cart.rows
  })
)(CartTable);
