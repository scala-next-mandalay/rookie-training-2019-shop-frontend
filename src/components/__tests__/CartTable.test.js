import React from 'react'
import renderer from 'react-test-renderer';
import Parent from '../Parent';
import CartTable from '../CartTable';
/*global expect*/

let container
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.outerHTML = '<body></body>';
  container = null;
});



// const valueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;



describe("CartTable component", () => {
  let cart = [{ id: 1, name: "a", quantity: 1, price: 1000 },
  { id: 2, name: "b", quantity: 1, price: 1000 }];
  it('matches the snapshot', () => {
    const CartTableSnapshot = renderer.create(
      <Parent><CartTable cart={cart} /></Parent>
    ).toJSON();
    expect(CartTableSnapshot).toMatchSnapshot();
  });
})