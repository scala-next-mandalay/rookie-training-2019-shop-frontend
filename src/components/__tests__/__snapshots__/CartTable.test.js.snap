// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartTable component matches the snapshot 1`] = `
Array [
  <div
    className="PrivateHiddenCss-smUp-11"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 makeStyles-paperMobile-5 MuiPaper-rounded"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <div
          className="MuiCardMedia-root makeStyles-imageMobile-4"
          style={
            Object {
              "backgroundImage": "url(\\"http://dg9c95ld8mycd.cloudfront.net/items/undefined\\")",
            }
          }
          title="a"
        />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <div
          className="MuiBox-root MuiBox-root-155"
        >
          <div
            className="MuiBox-root MuiBox-root-156"
          >
            Description:
            a
          </div>
          <div
            className="MuiBox-root MuiBox-root-157"
          >
            Qty:
            <div>
              1
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-158"
          >
            Price: @
            1000
             Ks
          </div>
          <div
            className="MuiBox-root MuiBox-root-159"
          >
            Total: 
             Ks
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation1 makeStyles-paperMobile-5 MuiPaper-rounded"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <div
          className="MuiCardMedia-root makeStyles-imageMobile-4"
          style={
            Object {
              "backgroundImage": "url(\\"http://dg9c95ld8mycd.cloudfront.net/items/undefined\\")",
            }
          }
          title="b"
        />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <div
          className="MuiBox-root MuiBox-root-160"
        >
          <div
            className="MuiBox-root MuiBox-root-161"
          >
            Description:
            b
          </div>
          <div
            className="MuiBox-root MuiBox-root-162"
          >
            Qty:
            <div>
              1
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-163"
          >
            Price: @
            1000
             Ks
          </div>
          <div
            className="MuiBox-root MuiBox-root-164"
          >
            Total: 
             Ks
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="PrivateHiddenCss-xsDown-9"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="makeStyles-girdDv-6"
      >
        <div
          className="MuiGrid-root makeStyles-gridRow-1 MuiGrid-container"
          variant="contained"
        >
          <div
            className="MuiGrid-root makeStyles-gridCell-2 MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div
              className="MuiBox-root MuiBox-root-165"
            >
              <span>
                Image
              </span>
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-gridCell-2 MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div
              className="MuiBox-root MuiBox-root-166"
            >
              <span>
                Description
              </span>
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-gridCell-2 MuiGrid-item MuiGrid-grid-xs-2"
          >
            <div
              className="MuiBox-root MuiBox-root-167"
            >
              <span>
                Price
              </span>
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-gridCell-2 MuiGrid-item MuiGrid-grid-xs-2"
          >
            <div
              className="MuiBox-root MuiBox-root-168"
            >
              <span>
                Quantity
              </span>
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-gridCell-2 MuiGrid-item MuiGrid-grid-xs-2"
          >
            <div
              className="MuiBox-root MuiBox-root-169"
            >
              <span>
                Total Price
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <hr
          className="MuiDivider-root"
        />
      </div>
      <div
        className="MuiGrid-root makeStyles-gridRow-1 MuiGrid-container"
      >
        <div
          className="MuiGrid-root makeStyles-gridCell-2 MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div
            className="MuiCardMedia-root makeStyles-imagePC-3"
            style={
              Object {
                "backgroundImage": "url(\\"http://dg9c95ld8mycd.cloudfront.net/items/undefined\\")",
              }
            }
            title="a"
          />
        </div>
        <div
          className="MuiGrid-root makeStyles-gridCell-2 MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div
            className="MuiBox-root MuiBox-root-175 priceTxt"
          >
            a
          </div>
        </div>
        <div
          className="MuiGrid-root makeStyles-gridCell-2 MuiGrid-item MuiGrid-grid-xs-2"
        >
          <div
            className="MuiBox-root MuiBox-root-176 priceTxt"
          >
            1000
          </div>
        </div>
        <div
          className="MuiGrid-root makeStyles-gridCell-2 MuiGrid-item MuiGrid-grid-xs-2"
        >
          <div
            className="MuiBox-root MuiBox-root-177"
          >
            <div
              className="MuiBox-root MuiBox-root-178"
            >
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline"
                onClick={[Function]}
              >
                <select
                  className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  value={1}
                >
                  <option
                    value={1}
                  >
                    1
                  </option>
                  <option
                    value={2}
                  >
                    2
                  </option>
                  <option
                    value={3}
                  >
                    3
                  </option>
                  <option
                    value={4}
                  >
                    4
                  </option>
                  <option
                    value={5}
                  >
                    5
                  </option>
                  <option
                    value={6}
                  >
                    6
                  </option>
                  <option
                    value={7}
                  >
                    7
                  </option>
                  <option
                    value={8}
                  >
                    8
                  </option>
                  <option
                    value={9}
                  >
                    9
                  </option>
                  <option
                    value={10}
                  >
                    10
                  </option>
                  <option
                    value={11}
                  >
                    11
                  </option>
                  <option
                    value={12}
                  >
                    12
                  </option>
                  <option
                    value={13}
                  >
                    13
                  </option>
                  <option
                    value={14}
                  >
                    14
                  </option>
                  <option
                    value={15}
                  >
                    15
                  </option>
                  <option
                    value={16}
                  >
                    16
                  </option>
                  <option
                    value={17}
                  >
                    17
                  </option>
                  <option
                    value={18}
                  >
                    18
                  </option>
                  <option
                    value={19}
                  >
                    19
                  </option>
                  <option
                    value={20}
                  >
                    20
                  </option>
                </select>
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiNativeSelect-icon"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root makeStyles-gridCell-2 MuiGrid-item MuiGrid-grid-xs-2"
        >
          <div
            className="MuiBox-root MuiBox-root-225"
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <hr
          className="MuiDivider-root"
        />
      </div>
      <div
        className="MuiGrid-root makeStyles-gridRow-1 MuiGrid-container"
      >
        <div
          className="MuiGrid-root makeStyles-gridCell-2 MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div
            className="MuiCardMedia-root makeStyles-imagePC-3"
            style={
              Object {
                "backgroundImage": "url(\\"http://dg9c95ld8mycd.cloudfront.net/items/undefined\\")",
              }
            }
            title="b"
          />
        </div>
        <div
          className="MuiGrid-root makeStyles-gridCell-2 MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div
            className="MuiBox-root MuiBox-root-226 priceTxt"
          >
            b
          </div>
        </div>
        <div
          className="MuiGrid-root makeStyles-gridCell-2 MuiGrid-item MuiGrid-grid-xs-2"
        >
          <div
            className="MuiBox-root MuiBox-root-227 priceTxt"
          >
            1000
          </div>
        </div>
        <div
          className="MuiGrid-root makeStyles-gridCell-2 MuiGrid-item MuiGrid-grid-xs-2"
        >
          <div
            className="MuiBox-root MuiBox-root-228"
          >
            <div
              className="MuiBox-root MuiBox-root-229"
            >
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline"
                onClick={[Function]}
              >
                <select
                  className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  value={1}
                >
                  <option
                    value={1}
                  >
                    1
                  </option>
                  <option
                    value={2}
                  >
                    2
                  </option>
                  <option
                    value={3}
                  >
                    3
                  </option>
                  <option
                    value={4}
                  >
                    4
                  </option>
                  <option
                    value={5}
                  >
                    5
                  </option>
                  <option
                    value={6}
                  >
                    6
                  </option>
                  <option
                    value={7}
                  >
                    7
                  </option>
                  <option
                    value={8}
                  >
                    8
                  </option>
                  <option
                    value={9}
                  >
                    9
                  </option>
                  <option
                    value={10}
                  >
                    10
                  </option>
                  <option
                    value={11}
                  >
                    11
                  </option>
                  <option
                    value={12}
                  >
                    12
                  </option>
                  <option
                    value={13}
                  >
                    13
                  </option>
                  <option
                    value={14}
                  >
                    14
                  </option>
                  <option
                    value={15}
                  >
                    15
                  </option>
                  <option
                    value={16}
                  >
                    16
                  </option>
                  <option
                    value={17}
                  >
                    17
                  </option>
                  <option
                    value={18}
                  >
                    18
                  </option>
                  <option
                    value={19}
                  >
                    19
                  </option>
                  <option
                    value={20}
                  >
                    20
                  </option>
                </select>
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiNativeSelect-icon"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root makeStyles-gridCell-2 MuiGrid-item MuiGrid-grid-xs-2"
        >
          <div
            className="MuiBox-root MuiBox-root-230"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;
